src/rgw/rgw_rest_s3.cc

```diff
@@ -3006,6 +3009,11 @@ RGWOp *RGWHandler_REST_Service_S3::op_post()
     if (action.compare("DeleteRolePolicy") == 0)
       return new RGWDeleteRolePolicy;
   }
+  if (this->isSTSenabled) {
+    RGWHandler_REST_STS sts_handler(auth_registry);
+    sts_handler.init(store, s, s->cio);
+    return sts_handler.get_op(store);
+  }
```

Comment:
why can't this stay under /sts? does that not work with boto?

Reply:
Yes, /sts wont work with boto, STS is served at a different endpoint. After discussing with Matt, we had decided to have it under the namespace as S3 and at the same endpoint for the time being.